src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -69,11 +72,59 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
                 object value;
                 if (obj.IsObject() && obj.IsArray() == false)
                 {
+                    //Note that for dynamic fields 'field.Name' will contain the real name of the field and not 'property'
```

Comment:
I don't see where this is actually happening, though?

Reply:
This is just a comment to explain the "magic" since i understand that it won't be clear to somebody reading the code. I have actually already implemented the logic more than a month ago when dealing with dynamic fields and there was very little i had to change to support the spatial fields. The value pased to the dynamic field is the createSpatialField(...) and it is later handled in the same code path as regular spatial fields.

Reply:
I'm not following this from the comment and the code, can you extend the comment?

Reply:
i'll try to make the comment clearer